src/hotspot/share/runtime/safepoint.cpp

```diff
@@ -706,7 +706,6 @@ void SafepointSynchronize::block(JavaThread *thread) {
   }
 
   JavaThreadState state = thread->thread_state();
-  assert(is_a_block_safe_state(state), "Illegal threadstate encountered: %d", state);
```

Comment:
Getting rid of this assert makes me a bit nervous, but I have to
get used to the fact that @pchilano is making the safepointing
sub-system simpler and simpler...

Reply:
I thought about keeping it but not only we have a single caller(SafepointMechanism::process()) which already has that assert as a guarantee, but also there is no way to add new callers without basically copying all the logic from SafepointMechanism::process() since we cannot just call SS::block().